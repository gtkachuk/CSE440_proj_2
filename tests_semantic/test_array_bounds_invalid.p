program testArrayBoundsInvalid;

class testArrayBoundsInvalid

BEGIN
   
VAR
   bad	: ARRAY[4..9] OF integer; 
   good	: ARRAY[78..123] OF integer;

FUNCTION testArrayBoundsInvalid;
BEGIN
   bad[3] := 67;
   bad[4] := 67;
   bad[5] := 67;
   bad[9] := 67;
   bad[10] := 67;
   bad[6] := 70;
   good[99] := bad[5]
END

END
.
