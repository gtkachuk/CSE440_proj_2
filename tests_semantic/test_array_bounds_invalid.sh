#!/bin/sh

. ../tests_lib.inc

test_diff_sem test_array_bounds_invalid.p
