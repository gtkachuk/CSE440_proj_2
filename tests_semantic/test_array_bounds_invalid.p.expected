ERROR:13:Array index 3 is out of the range (4,9)
ERROR:17:Array index 10 is out of the range (4,9)
Errors detected. Exiting.
